modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/AggregateProcess.java

```diff
@@ -67,18 +56,34 @@
      * @return aggregate Results
      */
     public static Results process(SimpleFeatureCollection features, String aggAttribute, Set<AggregationFunction> functions, Boolean singlePass, ProgressListener progressListener) throws ProcessException, IOException {
-        AggregateProcess process = new AggregateProcess();
-        return process.execute(features, aggAttribute, functions, singlePass, progressListener);
+        return process(features, aggAttribute, functions, null, singlePass, progressListener);
     }
+
+    public static Results process(SimpleFeatureCollection features, String aggAttribute, Set<AggregationFunction> functions,
+                                  List<String> groupByAttributes, Boolean singlePass, ProgressListener progressListener) throws ProcessException, IOException {
+        AggregateProcess process = new AggregateProcess();
+        return process.execute(features, aggAttribute, functions, singlePass, groupByAttributes, progressListener);
+     }
     
+    public Results execute(SimpleFeatureCollection features,String aggAttribute,Set<AggregationFunction> functions,
+                           boolean singlePass, ProgressListener progressListener) throws ProcessException, IOException {
+        return execute(features, aggAttribute, functions, singlePass, null, progressListener);
+    }
+
     @DescribeResult(name = "result", description = "Aggregation results (one value for each function computed)")
     public Results execute(
             @DescribeParameter(name = "features", description = "Input feature collection") SimpleFeatureCollection features,
             @DescribeParameter(name = "aggregationAttribute", min = 0, description = "Attribute on which to perform aggregation") String aggAttribute,
             @DescribeParameter(name = "function", description = "An aggregate function to compute. Functions include Count, Average, Max, Median, Min, StdDev, and Sum.", collectionType = AggregationFunction.class) Set<AggregationFunction> functions,
             @DescribeParameter(name = "singlePass", description = "If True computes all aggregation values in a single pass (this will defeat DBMS-specific optimizations)", defaultValue = "false") boolean singlePass,
+            @DescribeParameter(name = "groupByAttributes", description = "List of group by attributes", collectionType = String.class) List<String> groupByAttributes,
```

Comment:
Is it worth making this one optional - to preserve backwards compatibility? Probably not ...